packages/react-devtools-scheduling-profiler/src/import-worker/import.worker.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import 'regenerator-runtime/runtime';
+
+import type {TimelineEvent} from '@elg/speedscope';
+import type {ReactProfilerData} from '../types';
+
+import preprocessData from './preprocessData';
+import {readInputData} from './readInputData';
+import InvalidProfileError from './InvalidProfileError';
+
+declare var self: DedicatedWorkerGlobalScope;
+
+type ImportWorkerInputData = {|
+  file: File,
+|};
+
+export type ImportWorkerOutputData =
+  | {|status: 'SUCCESS', processedData: ReactProfilerData|}
+  | {|status: 'INVALID_PROFILE_ERROR', error: Error|}
+  | {|status: 'UNEXPECTED_ERROR', error: Error|};
```

Comment:
Ideally we'll be able to use `{|status: 'ERROR', error: Error | InvalidProfileError|}` and differentiate between the 2 cases with an `instanceof` check, but we can't do that here because `InvalidProfileError` instances revert to being plain `Error` instances when passed to the main thread.